.github/workflows/package-tests.yml

```diff
@@ -52,18 +52,48 @@ jobs:
           then
             echo "::error::pre-commit hooks failed for ${{ matrix.package_name }}" && exit 1
           fi
+
+  tests:
+    name: Test | ${{ matrix.package_name }}
+    needs: package-filter
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJson(needs.package-filter.outputs.matrix) }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.9'
+      - name: Install Poetry
+        uses: abatilo/actions-poetry@v2
+      - name: Install pre-commit
```

Comment:
so do we need the precommit hooks since we run them separatedly?

Reply:
Fixed.